lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -231,6 +232,10 @@ void EfficientGlobalOptimization::run()
   // use the provided kriging result at first iteration
   KrigingResult metaModelResult(krigingResult_);
 
+  // EGO expects KrigingAlgorithm to estimate hyperparameters with a single-start OptimizationAlgorithm.
+  TNC krigingSolver;
+  krigingSolver.setIgnoreFailure(true);
```

Comment:
do you know why we need this ? in principle it should be only used when the algorithm does not converge but the last point is actually a good value, I know I set it up for the arma factory but I dont know if its relevant for kriging

Reply:
No, I do not know why we need it. This is what happens in1.16, though. Perhaps I can try to remove it and see what happens?

Reply:
Removing it causes unit tests failure (see comment https://github.com/openturns/openturns/pull/1827#issuecomment-826997247 below).